File: ObjC/SIALoggerSample/SIALoggerSample/ViewController.m
//
//  ViewController.m
//  SIALoggerSample
//
//  Created by Alexander Ivlev on 31/05/16.
//  Copyright © 2016 Alexander Ivlev. All rights reserved.
//

#import "ViewController.h"
#import <SIALogger/SIALogger.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  
  [SIALogConfig setOutputs:@[ [SIALogColoredConsoleOutput new] ]];
  
  [SIALogConfig setMaxLogLevel: SIALogLevels.Info];

  SIALogTrace(@"1 trace no show");
  
  SIALogInfo(@"1 info");
  SIALogWarning(@"1 warning");
  SIALogError(@"1 error");
  
  [SIALogConfig setMaxLogLevel: SIALogLevels.Error];
  SIALogTrace(@"2 trace no show");
  SIALogInfo(@"2 info no show");
  SIALogWarning(@"2 warning no show");
  SIALogError(@"2 error");
  
  [SIALogConfig setMaxLogLevel: SIALogLevels.Trace];
  SIALogTrace(@"3 trace");
  
  SIALogTrace(@"4 trace");
  SIALogInfo(@"4 info");
  SIALogWarning(@"4 warning");
  SIALogError(@"4 error");
  
  SIALogTraceIf(true, @"Trace if");
  SIALogInfoIf(false, @"Info if no show");
  SIALogWarningIf(true, @"Warning if");
  SIALogErrorIf(false, @"Error if no show");
  
  if (SIALogTraceIf(false, @"Trace if no show")) {
    SIALogInfo(@"By trace no show");
  }
  
  if (SIALogInfoIf(true, @"Info if")) {
    SIALogInfo(@"By info");
  }
  
  if (SIALogWarningIf(false, @"Warning if no show")) {
    SIALogInfo(@"By warning no show");
  }
  
  if (SIALogErrorIf(true, @"Error if")) {
    SIALogInfo(@"By error");
  }
  
  SIALogAssertMsg(true, @"no show");
  SIALogAssertMsg(false, @"assert");
}

@end

File: Swift/SIALoggerSample/SIALoggerSample/ViewController.swift
//
//  ViewController.swift
//  SIALoggerSample
//
//  Created by Alexander Ivlev on 31/05/16.
//  Copyright © 2016 Alexander Ivlev. All rights reserved.
//

import UIKit
import SIALogger

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()
    
    SIALogConfig.maxLogLevel = SIALogLevel.Info
    SIALog.Trace("1 trace no show")
    SIALog.Info("1 info")
    SIALog.Warning("1 warning")
    SIALog.Error("1 error")
    
    SIALogConfig.maxLogLevel = SIALogLevel.Error
    SIALog.Trace("2 trace no show")
    SIALog.Info("2 info no show")
    SIALog.Warning("2 warning no show")
    SIALog.Error("2 error")
    
    SIALogConfig.maxLogLevel = SIALogLevel.Trace
    SIALog.Trace("3 trace")
    
    SIALog.Trace("4 trace")
    SIALog.Info("4 info")
    SIALog.Warning("4 warning")
    SIALog.Error("4 error")
    
    
    SIALog.TraceIf(true, msg: "Trace if")
    SIALog.InfoIf(false, msg: "Info if no show")
    SIALog.WarningIf(true, msg: "Warning if")
    SIALog.ErrorIf(false, msg: "Error if no show")
    
    if (SIALog.TraceIf(false, msg: "Trace if no show")) {
      SIALog.Info("By trace no show")
    }
    
    if (SIALog.InfoIf(true, msg: "Info if")) {
      SIALog.Info("By info")
    }
    
    if (SIALog.WarningIf(false, msg: "Warning if no show")) {
      SIALog.Info("By warning no show")
    }
    
    if (SIALog.ErrorIf(true, msg: "Error if")) {
      SIALog.Info("By error")
    }
    
    SIALog.Assert(true, msg: "no show")
    SIALog.Assert(false, msg: "assert")
  }
}

